WatchDogIntelliJPlugin/WatchDog/src/nl/tudelft/watchdog/intellij/logic/ui/listeners/JUnitListener.java

```diff
@@ -1,31 +1,39 @@
 package nl.tudelft.watchdog.intellij.logic.ui.listeners;
 
+import com.intellij.execution.Location;
 import com.intellij.execution.testframework.AbstractTestProxy;
 import com.intellij.execution.testframework.TestStatusListener;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.project.ProjectManager;
+import com.intellij.psi.search.GlobalSearchScope;
 import nl.tudelft.watchdog.intellij.logic.InitializationManager;
 import nl.tudelft.watchdog.intellij.logic.interval.intervaltypes.JUnitInterval;
 import nl.tudelft.watchdog.intellij.logic.ui.EventManager;
 import nl.tudelft.watchdog.core.logic.ui.events.JUnitEvent;
-import nl.tudelft.watchdog.core.util.WatchDogGlobals;
+import nl.tudelft.watchdog.intellij.util.WatchDogUtils;
 
 /** A listener to the execution of Junit test via the IDE. */
 public class JUnitListener extends TestStatusListener {
 
-    private EventManager eventManager;
-
-    public JUnitListener() {
-        if(WatchDogGlobals.isActive) {
-            this.eventManager = InitializationManager.getInstance().getEventManager();
-        }
-    }
-
     @Override
     public void testSuiteFinished(AbstractTestProxy abstractTestProxy) {
-        if(!WatchDogGlobals.isActive) return;
-        if(eventManager == null) {
-            this.eventManager = InitializationManager.getInstance().getEventManager();
+        if(!WatchDogUtils.isWatchDogActive(getProject(abstractTestProxy))) {
+            return;
         }
+        EventManager eventManager = InitializationManager.getInstance(getProject(abstractTestProxy).getName()).getEventManager();
         JUnitInterval interval = new JUnitInterval(abstractTestProxy);
         eventManager.update(new JUnitEvent(interval));
     }
+
+    private Project getProject(AbstractTestProxy test) {
```

Comment:
Actually, this should never return null. But it needs `return null;` because of the language requirement.